### PR TITLE
Fixed geo-dashoard stream connection issue and broken pipe issue

### PR DESCRIPTION
**Pull request by lasanthaDLPDS**

### CODE REVIEW

**Review by rasika — APPROVED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js [hunk 1]
```diff
@@ -17,30 +17,38 @@
...
-        if (tokenPair) {
-            token = tokenPair.accessToken;
+        if (context.user.domain == "carbon.super") {
+            tokenPair = jwtClient.getAccessToken(resp[0], resp[1], context.user.username,"default", {});
+            if (tokenPair) {
+                token = tokenPair.accessToken;
+                wsEndpoint = devicemgtProps["wssURL"].replace("https", "wss") + "/secured-websocket/";
+            }
+        } else {
+            tokenPair = jwtClient.getAccessToken(resp[0], resp[1], context.user.username + "@" + context.user.domain,"default", {});
```
Comment (madhawap) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js:44]: This line seems exceeding the right margin column which should be 110. Please reformat it.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js [hunk 2]
```diff
@@ -17,30 +17,38 @@
...
+            if (tokenPair) {
+                token = tokenPair.accessToken;
+                wsEndpoint = devicemgtProps["wssURL"].replace("https", "wss") + "/secured-websocket/t/"+context.user.domain+"/";
```
Comment (madhawap) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js:47]: This line also exceeding the right margin column which should be 110. Please reformat it.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/websocket.js [hunk 3]
```diff
@@ -518,26 +561,20 @@ var webSocketOnAlertMessage = function processMessage(message) {
         var json = $.parseJSON(message.data);
         if (json.messageType == "Alert") {
             processAlertMessage(json);
-        } else {
+        }else {
```
Comment (madawas) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/websocket.js:564]: @lasanthaDLPDS Please fix the formatting issue here.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js (hunk 1 above)
Comment (lasanthaDLPDS) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js:44]: This is Fixed c86c976fe6ca2d0e8b534b7efc441713c7e529b7

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js (hunk 2 above)
Comment (lasanthaDLPDS) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/geo-dashboard.js:47]: This is fixed c86c976fe6ca2d0e8b534b7efc441713c7e529b7.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/websocket.js (hunk 3 above)
Comment (lasanthaDLPDS) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/websocket.js:564]: This is fixed c86c976fe6ca2d0e8b534b7efc441713c7e529b7.

**Pull request merged**